src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -42,14 +42,33 @@ internal static string Normalize(string path)
             return result;
         }
 
-        private static void GetFullPathName(string path, ref ValueStringBuilder builder)
+        internal static string Normalize(ReadOnlySpan<char> path)
```

Comment:
Would it be better for Normalize to take the ValueStringBuilder as argument (non-zero terminated) and zero terminate it before the PInvoke?

It would also address my concern of zero-terminated `Span<char>`s  leaking through the system.

Reply:
I'll give it a look and get back to you shortly.